syntax = "proto3";

package tmapreduce;

option cc_generic_services = true;

enum WorkerState {
  UNKNOWN = 0;
  INIT = 1;
  IDLE = 2;
  WAIT2MAP = 3;
  MAPPING = 4;
  WAIT2REDUCE = 5;
  REDUCING = 6;
  WAIT2SUBMITMAP = 7;
  WAIT2SUBMITREDUCE = 8;
  CLOSE = 9;
}